curriculum/challenges/english/25-front-end-development/lab-pyramid-generator/66f2836c459cfb16ae76f24f.md

```diff
@@ -0,0 +1,78 @@
+---
+id: 66f2836c459cfb16ae76f24f
+title: Build a Pyramid Generator
+challengeType: 14
+dashedName: lab-pyramid-generator
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should have a function named `pyramid` that takes three arguments.
+1. The first argument should be a string representing the pattern character to repeat in your pyramid.
+1. The second argument should be an integer representing the number of rows in the pyramid.
+1. The third argument should be a Boolean value.
+1. The `pyramid` function should return a string in which the pattern character is repeated and arranged to form a pyramid having the vertex facing upwards when the third argument is `false`.
+1. When the third argument is `true` the pyramid should have the vertex facing downwards.
+1. The vertex row should have a single pattern character, and each other row should have two pattern characters more than the previous one.
+1. Each row should start with a number of spaces sufficient to put the center character of each row in the same column.
+
+For example, calling `pyramid('o', 4, false)` should give this output:
+
+```js
+
+   o
+  ooo
+ ooooo
+ooooooo
+```
+
+# --hints--
+
+You should have a function named `pyramid` that takes three arguments.
```

Comment:
There should be a check that there are actually 3 parameters.
Currently the test is passing with any number of arguments.

Reply:
That's not an easy thing to test, but as long as the other tests pass, the camper is using the three parameters, so I don't think it's necessary, it would be a regex, and those can become quite rigid